packages/runtime/src/runtimes/react/root-style-registry.tsx

```diff
@@ -1,62 +1,15 @@
 'use client'
 
-import createCache, { type EmotionCache } from '@emotion/cache'
-import { cache } from '@emotion/css'
 import { type PropsWithChildren, createContext, useContext } from 'react'
 
-const CacheContext = createContext(cache)
+import { type BaseStyleCache, type StyleTagProps, createStyleCache } from './lib/style-cache/base'
+
+// default cache context with a sentinel key signaling misconfiguration
+const CacheContext = createContext(createStyleCache({ key: 'msft-err' }))
 
 const DEFAULT_CSS_RESET_ENABLED = true
 const CSSResetEnabledContext = createContext(DEFAULT_CSS_RESET_ENABLED)
 
-export type StyleCache = EmotionCache & {
-  /**
-   * Flush the inserted styles.
-   * @returns A list of class names and the corresponding CSS rules that were flushed.
-   */
-  flush: () => { classNames: string[]; css: string }
-}
-
-export const createRootStyleCache = ({ key }: { key?: string } = {}): StyleCache => {
```

Comment:
Split up into [`createStyleCache`](https://github.com/makeswift/makeswift/pull/1143/files#diff-acd61ad47550fdfd4de0a241194a8e111ff1c721cf34f630b3b070ad382d1b08R5), [`createFlushableStyleCache`](https://github.com/makeswift/makeswift/pull/1143/files#diff-9ba6102f1dc18c2bfe2947e447e66fc60b0e3d6fc0024885c84e7e610a9e07ffR11) and [`createStreamingStyleCache`](https://github.com/makeswift/makeswift/pull/1143/files#diff-790c9082df4ceccb4062b884ad48538911f18eab8d2c462d1e2f047383e57873R8).